x-pack/plugins/alerting/public/plugin.ts

```diff
@@ -59,7 +61,28 @@ export class AlertingPublicPlugin implements Plugin<PluginSetupContract, PluginS
       ruleTypeId: string,
       handler: AlertNavigationHandler
     ) => {
-      this.alertNavigationRegistry!.register(applicationId, ruleTypeId, handler);
+      // if the ruleTypes not fetched yet than to the API call and cache the results
+      if (!this.ruleTypes) {
+        this.ruleTypes = await loadRuleTypes({ http: core.http });
+        if (!this.ruleTypes) {
+          // eslint-disable-next-line no-console
+          console.log(
```

Comment:
What kind of experience does this cause for the user? Does the rule type not show up in the creation/edit flow?

Reply:
The button View in app will be disable and the proper message will be in the browser console

Reply:
Gotcha, makes sense!